cdap-ui/app/cdap/components/PipelineConfigurations/Store/ActionCreator.js

```diff
@@ -34,6 +43,37 @@ const applyRuntimeArgs = () => {
   });
 };
 
+// Filter certain preferences from being shown in the run time arguments 
+// They are being represented in other places (like selected compute profile).
+const getFilteredRuntimeArgs = () => {
+  let runtimeArgs = PipelineConfigurationsStore.getState().runtimeArgs;
+  let modifiedRuntimeArgs = {};
+  let pairs = [...runtimeArgs.pairs];
+  pairs = pairs.filter(pair => (RUNTIME_ARGS_TO_SKIP_DURING_DISPLAY.indexOf(pair.key) === -1));
+  if (!pairs.length) {
+    pairs.push(getDefaultKeyValuePair());
+  }
+  modifiedRuntimeArgs.pairs = pairs;
+  return modifiedRuntimeArgs;
+};
+
+// While adding runtime argument make sure to include the excluded preferences
```

Comment:
aren't the preferences already included by default (by backend)? why do the UI need to pass it as runtime arg as well?

Reply:
So this is a little bit different.

When displaying runtime arguments in the configure modeless we want to exclude properties like `logical.start.time` and `system.profiles.name`. But while making changes on the key-value pairs we want to make sure we include those while saving to the store.

This function is provided for `onKeyValuesChange` to `KeyValuePair` in `RuntimeArgsPairs.js` module.